lib/Runtime/Provisioner/Provisioner.cpp

```diff
@@ -473,13 +473,15 @@ Error Provisioner::provision(DAGListTy &networks, Module &module,
 
       // Transfer weight to all devices needed.
       for (const auto &device : placeholderToDevicManager[PH]) {
-        std::promise<Error> transferPromise;
+        std::promise<void> transferPromise;
+        Error transferError = Error::empty();
         auto done = transferPromise.get_future();
         devices_[device]->transferStaticPlaceholderToDevice(
-            PH, weight, [&transferPromise](Error err) {
-              transferPromise.set_value(std::move(err));
+            PH, weight, [&transferPromise, &transferError](Error err) {
+              transferError = std::move(err);
+              transferPromise.set_value();
             });
-        RETURN_IF_ERR(done.get());
+        RETURN_IF_ERR(std::move(transferError));
```

Comment:
same

Reply:
@jackm321 Hmm? I don't think RETURN_IF_ERR supports output of ERR_TO_BOOL looking at it's implementation it actaully expects GlowError.

Reply:
Oh sorry, I misread this line as being similar to below! Incidentally I think this move isn't needed since RETURN_IF_ERR forwards `transferError` but I could be wrong

Reply:
Nope, you are right. pushed update.